### ISSUE TITLE
Migration fails on Ganeti 3.0.1 with security_model=user

### ISSUE THREAD

**Issue by derryaire** [ganeti/ganeti#1601]
Hello there,

It seems migration is impossible (VM freezes) with _security_model=user_ on _Ubuntu 20.04/Qemu 4.2.1/Ganeti 3.0.1_ with this PPA: https://launchpad.net/~pkg-ganeti-devel/+archive/ubuntu/lts

The VM freezes after being migrated to the target node.

Setting _security_domain=,security_model=none_ allows for migration to work again - but really is not such a great option.

The same configuration works without problems on _Ubuntu 18.04/Qemu 2.11.1/Ganeti 3.0.1_ with Apollon's PPA.

There is very little in the way of logging apart from this:

**Node1:**

```
==> /var/log/ganeti/kvm/testing.log <==
qemu-system-x86_64: Could not reopen file: Permission denied

==> /var/log/ganeti/node-daemon.log <==
2021-07-01 13:02:37,922: ganeti-noded pid=178241 INFO 10.1.1.48:40956 POST /instance_finalize_migration_dst HTTP/1.1 200
2021-07-01 13:02:37,987: ganeti-noded pid=178242 INFO 10.1.1.48:40960 POST /upload_file_single HTTP/1.1 200
```

**Node2:**
```
==> /var/log/ganeti/kvm/testing.log <==
qemu-system-x86_64: terminating on signal 15 from pid 188338 (/usr/bin/python3)

==> /var/log/ganeti/node-daemon.log <==
2021-07-01 13:02:37,869: ganeti-noded pid=188338 INFO 10.1.1.48:39282 POST /instance_finalize_migration_src HTTP/1.1 200
2021-07-01 13:02:37,993: ganeti-noded pid=188344 INFO 10.1.1.48:39290 POST /upload_file_single HTTP/1.1 200
2021-07-01 13:02:38,000: ganeti-noded pid=188345 INFO RunCmd lvs --noheadings '--separator=|' --units=k --nosuffix -ovg_name,lv_name,lv_attr,lv_kernel_major,lv_kernel_minor,vg_extent_size,stripes,devices
```

Update: this happens while using DRBD for disk replication

**Comment by saschalucas** [comment]
Thanks for your report. I can confirm the issue on Ubuntu Focal.

I tried to confirm this on Debian Bullseye, too. However a little yak shave happened during initial preparation of a my own test environment, which I haven't fully resolved yet:

* #1602 
* it seems in Bullseye the Linux Bridge's MAC is no longer random. All the nodes of my testcluster ended up with the same MAC
```
ip l add dev gnt-br0 type bridge
ip -br l l dev gnt-br0
gnt-br0          DOWN           4a:cd:94:30:27:72 <BROADCAST,MULTICAST> 
```

If I would guess, this problem could be related to @apoikos fix for disk hotplug (commit 6d6adbb7187b4130c74e877e114222f039d35e11).

**Comment by saschalucas** [comment]
Just a small update. This BUG also affects Debian Bullseye, however the included qemu-5.2 is a little bit more verbose:
```
==> /var/log/ganeti/kvm/test.vm.log <==
kvm: Could not open '/var/run/ganeti/instance-disks/test.vm:0': Permission denied
```
As I assumed, it tries to (re)open the disk, just like with hotplug. If that's the case, a fix should be simple.

@rbott: I assume, you do not test live migration with `security_model=user` in your qa-setup?

@apoikos: is it worth to open a Debian bug (I have tested from source, but see no reason, why the Debian package should not be affected, too)?

**Comment by derryaire** [comment]
Thanks @saschalucas . Your debug output was quite helpful. _/var/run/ganeti/instance-disks/test.vm:0_ seems to be a link to _/dev/drbdX_ with the group id _disk_.
After adding my user from security_domain to the group _disk_ as a workaround I can migrate again.
